cloud-azure/src/test/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResourcesTest.java

```diff
@@ -85,77 +116,109 @@ public void testVariantShouldReturnAzurePlatform() {
+        return CloudContext.Builder.builder()
+                .withId(1L)
+                .withName("envName")
+                .withCrn("crn:cdp:environments:us-west-1:dummyUser:environment:randomGeneratedResource")
+                .withPlatform("AZURE")
+                .withVariant("AZURE")
+                .withLocation(location(region("dummyRegion")))
+                .withUserName("dummyUser")
+                .withAccountId("dummyAccountId")
+                .build();
+    }
+
+    private void initExceptionConversion() {
+        when(azureUtils.convertToCloudConnectorException(any(Exception.class), any(String.class)))
+                .thenAnswer(invocation -> new CloudConnectorException(invocation.getArgument(0, Exception.class)));
+    }
+
+    private void initActionFailedExceptionConversion() {
+        when(azureUtils.convertToActionFailedExceptionCausedByCloudConnectorException(any(Exception.class), any(String.class)))
+                .thenAnswer(invocation -> new Retry.ActionFailedException(new CloudConnectorException(invocation.getArgument(0, Exception.class))));
+    }
+
     @Test
     public void testExceptionIsThrownWhenNotIsSingleResourceGroup() {
         DiskEncryptionSetCreationRequest requestedSet = new DiskEncryptionSetCreationRequest.Builder()
-                .withCloudCredential(new CloudCredential())
-                .withRegion(Region.region("dummyRegion"))
-                .withEnvironmentName("dummyEnvName")
-                .withEnvironmentId(1L)
+                .withCloudCredential(cloudCredential)
+                .withCloudContext(cloudContext)
                 .withSingleResourceGroup(false)
                 .withResourceGroupName("dummyResourceGroup")
                 .withTags(new HashMap<>())
                 .withEncryptionKeyUrl("https://dummyVaultName.vault.azure.net/keys/dummyKeyName/dummyKeyVersion")
                 .build();
-        when(azureClientService.getClient(any(CloudCredential.class))).thenReturn(azureClient);
+        when(azureClientService.createAuthenticatedContext(cloudContext, cloudCredential)).thenReturn(authenticatedContext);
+        when(authenticatedContext.getParameter(AzureClient.class)).thenReturn(azureClient);
+        initExceptionConversion();
 
-        assertEquals(assertThrows(IllegalArgumentException.class, () -> underTest.createDiskEncryptionSet(requestedSet)).getMessage(),
+        verifyException(IllegalArgumentException.class, () -> underTest.createDiskEncryptionSet(requestedSet),
                 "Customer Managed Key Encryption for managed Azure disks is supported only if the CDP resources " +
                         "are in the same resource group as the vault.");
     }
 
+    private <T extends Throwable> void verifyException(Class<T> expectedType, Executable executable, String messageExpected) {
+        CloudConnectorException cloudConnectorException = assertThrows(CloudConnectorException.class, executable);
```

Comment:
hint: assertj has fancy methods for exception validation

Reply:
I know, but some folks (notably @doktoric 😄 ) prefer the JUnit `assertThrows()` over AssertJ `assertThatCode()`.